lib/ra10ke.rb

```diff
@@ -35,35 +36,38 @@ def initialize(*args)
             end
 
             if puppet_module.class == R10K::Module::Git
-              remote = puppet_module.instance_variable_get(:@remote)
-
               # use helper; avoid `desired_ref`
               # we do not want to deal with `:control_branch`
               ref = puppet_module.version
               next unless ref
 
+              remote = puppet_module.instance_variable_get(:@remote)
               remote_refs = Git.ls_remote(remote)
 
               # skip if ref is a branch
               next if remote_refs['branches'].key?(ref)
 
-              ref_type = 'sha'    if ref.match(/^[a-z0-9]{40}$/)
-              ref_type = 'tag'    if remote_refs['tags'].key?(ref)
-              case ref_type
-              when 'sha'
+              if remote_refs['tags'].key?(ref)
+                # there are too many possible versioning conventions
+                # we have to be be opinionated here
+                # so semantic versioning (vX.Y.Z) it is for us
+                # as well as support for skipping the leading v letter
+                tags = remote_refs['tags'].keys
+                latest_tag = tags.map do |tag|
+                  begin
+                    Semverse::Version.new tag
+                  rescue Semverse::InvalidVersionFormat
+                    # ignore tags that do not comply to semver
+                    nil
+                  end
+                end.select { |tag| !tag.nil? }.sort.last.to_s.downcase
+                latest_ref = tags.detect { |tag| tag.downcase == latest_tag || tag.downcase == latest_tag[1..-1] }
```

Comment:
In retrospect, `tag.downcase == latest_tag[1..-1]` here could use an extra `latest_tag[1] == 'v'` check.